### PR TITLE
GitHub pages website for JDL

### PR DESCRIPTION
**Pull request by tfennelly**
**Description**

See http://tfennelly.github.io/jenkins-design-language/ for a demo of it pushed to my fork ... we will push to this repo if we merge.

* Generated via a static site generator named [HarpJS](http://harpjs.com/).
* Use JDL themes for site styling.
* Embeds the StoryBook in the `./docs` page.
* Added some docs on Button and EmptyState.
* Site is generated and published as part of executing `npm publish` (`prepublish` script) i.e. nothing extra to do.

__NOTE__: Nobody is suggesting this is "done", but is a lot better than what we have atm. Others with more taste can add more polish.

**Submitter checklist**
- [x] Change is code complete and matches issue description.
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests. N/A.

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

File: .flowconfig [hunk 1]
```diff
@@ -8,6 +8,9 @@
 # Yay for dependencies that are bloated with tests, and .json files that aren't JSON
 .*/node_modules/cjson/test.*
 
+.*/node_modules/config-chain/test/broken.json
+.*/node_modules/npmconf/test/fixtures/package.json
```
Comment (tfennelly) [inline .flowconfig]: @sophistifunk is it ok for me to do this? Build fails without it. The changes I made drag in some new deps and flow doesn't seem to like them. One of the new deps has a file named "broken.json" that has badly formatted json and causes flow to barf. Was probably from a test and got published.

Comment (sophistifunk) [inline .flowconfig]: Yes,  that's the same reason I had to add cjson. The concept of separate "dev" and "dist" filesets is apparently a novelty in node-land.

**Comment by sophistifunk** [comment]
What's `.ejs`?

**Comment by tfennelly** [comment]
@sophistifunk [ejs](http://www.embeddedjs.com/) is a templating framework supported by [harpjs](http://harpjs.com).

File: site-publish.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+git remote add webpages git@github.com:tfennelly/jenkins-design-language.git
+git fetch --depth=1 webpages gh-pages
+
+git add --all
+git commit -m "webpages"
+git merge --no-edit -s ours remotes/webpages/gh-pages
+
+git push webpages master:gh-pages
+
+rm -rf .git
```
Comment (michaelneale) [inline site-publish.sh]: is this needed when it is cleaned out the next time you run it? (I presume you leave it there so people can inspect?)

File: package.json [hunk 3]
```diff
@@ -34,6 +38,7 @@
     "url": "https://github.com/jenkinsci/jenkins-design-language.git"
   },
   "dependencies": {
+    "harp": "^0.20.3",
```
Comment (michaelneale) [inline package.json]: will this be used by other stuff in blue ocean? (will it be brought along as a transitive dep?)

**Comment by michaelneale** [comment]
I like it - main Q was around bringing harpjs on (fine for site, but will it be used elsewhere?) 
(comments on code lines)

File: site-publish.sh (hunk 2 above)
Comment (tfennelly) [inline site-publish.sh]: No, `website-build` is in `.gitignore` and the site is generated into it i.e. the generated stuff is never committed to the repo. We just set up a temp remote onto the repo inside this dir and push the generated website to the `gh-pages` branch and then remove the remote.

File: package.json (hunk 3 above)
Comment (tfennelly) [inline package.json]: Ooops yes ... that should be a `dev` dep.

**Comment by michaelneale** [comment]
:bee:

**Pull request merged**